src/Interpreter.js

```diff
@@ -39,22 +42,30 @@ export default class Interpreter {
         this.programCounter = 0;
     }
 
-    run(program: Program): void {
+    run(program: Program): Promise<void> {
         this.program = program;
         this.programCounter = 0;
         this.isRunning = true;
-        this.continueRun();
+        return new Promise((resolve, reject) => {
+            this.continueRun(resolve, reject);
+        });
     }
 
-    continueRun(): void {
+    continueRun(resolve: any, reject: any): void {
         if (this.isRunning) {
             if (this.atEnd()) {
                 this.isRunning = false;
+                this.onRunningStateChange({isRunning: this.isRunning, activeStep: null});
+                resolve();
             } else {
+                this.onRunningStateChange({isRunning: this.isRunning, activeStep: this.programCounter});
                 this.step().then(() => {
-                    this.continueRun();
+                    this.continueRun(resolve, reject);
                 });
+                // TODO: reject run promise if step promise rejected
             }
```

Comment:
Address the TODO comment by calling `reject` when the `step()` Promise is rejected:

````
} else {
    this.onRunningStateChange({isRunning: this.isRunning, activeStep: this.programCounter});
    this.step().then(() => {
        this.continueRun(resolve, reject);
    }, () => {
        // Reject the run Promise when the step Promise is rejected
        reject();
    });
}
````

And add a test case to `Interpreter.test.js` to verify that the run Promise is rejected.

For an example of asserting a Promise rejection, see: https://github.com/codelearncreate/c2lc-coding-environment/blob/6d770ca4e32482e6d657d1b1497c941fdf0bdf19/src/Interpreter.test.js#L95-L100